features/google/src/google.jsx

```diff
@@ -17,17 +17,38 @@ import Overview from "./views/overview/overview.jsx";
 import ImportView from "./views/import/import.jsx";
 
 import "./styles.css";
+import {
+    PolyImportContext,
+    PolyImportProvider,
+    INITIAL_HISTORY_STATE,
+} from "@polypoly-eu/poly-look";
+import GoogleAccount from "./model/google-account.js";
+import { dataImporters } from "./model/importer.js";
 
 const Google = () => {
-    const { pod } = useContext(GoogleContext);
-    const renderSplash = () => "Loading";
+    const { pod, isLoading } = useContext(GoogleContext);
+
+    const { files } = useContext(PolyImportContext);
+
+    function determineRoute() {
+        if (files.length > 0)
+            return (
+                <Redirect
+                    to={{
+                        pathname: "/overview",
+                        state: INITIAL_HISTORY_STATE,
+                    }}
+                />
+            );
+        else return <Redirect to={{ pathname: "/import" }} />;
+    }
 
     return (
         <div className="google poly-theme poly-theme-light">
-            {pod ? (
+            {pod && files && !isLoading && (
                 <Switch>
                     <Route exact path="/">
-                        <Redirect to={{ pathname: "/overview" }} />
+                        {determineRoute()}
```

Comment:
Unless you want to introduce more logic later on, `files.length` will be > 0, so this will be equivalent to the previous code except for the state. If what you're looking for is encapsulating that Redirect a bit more properly, that's OK; otherwise is it possible that `files` shouldn't be checked?

Reply:
Did you check what I said about `files` here?

Reply:
Yeah files existence and length have different  meaning in this scope. So this is meant to be like that. Files existance = we finished loading, files length = there are/n't files

Reply:
OK, TIL that's one of those JS quirks, `files` being truish even when empty.